pkg/api/handlers.go

```diff
@@ -305,7 +306,7 @@ func NewQuerierHandler(
 	router.Path(path.Join(promPrefix, "/query_exemplars")).Methods("GET", "POST").Handler(exemplarsQueryStats.Wrap(promRouter))
 	router.Path(path.Join(promPrefix, "/labels")).Methods("GET", "POST").Handler(labelsQueryStats.Wrap(promRouter))
 	router.Path(path.Join(promPrefix, "/label/{name}/values")).Methods("GET").Handler(labelsQueryStats.Wrap(promRouter))
-	router.Path(path.Join(promPrefix, "/series")).Methods("GET", "POST", "DELETE").Handler(seriesQueryStats.Wrap(promRouter))
+	router.Path(path.Join(promPrefix, "/series")).Methods("GET", "POST", "DELETE").Handler(limiter.MemoryTrackerMiddleware{}.Wrap(seriesQueryStats.Wrap(promRouter)))
```

Comment:
Is there a reason why we need to do this here rather than in `pkg/frontend/querymiddleware/roundtrip.go`'s `newQueryTripperware`?

If not, I'd suggest moving this to that file and moving the middleware to the `querymiddleware` package as well.